### PR TITLE
chore(verifytx): Updating VerifyTx to Cache between Transactions

### PR DESCRIPTION
**Pull request by davidterpay**

### CODE REVIEW

File: block/base/abci.go [hunk 1]
```diff
@@ -122,12 +122,20 @@ func (l *BaseLane) ProcessLane(
 	return next(ctx, proposal)
 }
 
-// AnteVerifyTx verifies that the transaction is valid respecting the ante verification logic of
+// VerifyTx verifies that the transaction is valid respecting the ante verification logic of
 // of the antehandler chain.
-func (l *BaseLane) AnteVerifyTx(ctx sdk.Context, tx sdk.Tx, simulate bool) (sdk.Context, error) {
+func (l *BaseLane) VerifyTx(ctx sdk.Context, tx sdk.Tx, simulate bool) error {
```
Comment (nivasan1) [inline block/base/abci.go:127]: Is this necessary? Notice, every time we cache the ctx we add a significant over-head to memory consumption (this  is one of the most costly things to do in SDK). Should the caching logic not be handled by the caller?

Comment (nivasan1) [reply block/base/abci.go:127]: If we do intend to keep this logic here, maybe make note exactly what the side-effect of the fn call is

**Review by nivasan1 — APPROVED** [review]
lgtm with nit

File: block/base/abci.go (hunk 1 above)
Comment (davidterpay) [reply block/base/abci.go:127]: Eh, probably not entirely necessary but if you think about execution from a deliver tx perspective, the entire context is cached before the tx is executed and state is only modified if the tx passes the expected checks.

Comment (davidterpay) [reply block/base/abci.go:127]: I think the atomicity of checking a tx is a nice feature, ill add a note that this is the behavior they should expect. In the case where they dont want this, its a super easy change

**Pull request merged**